test/api/pyapi/testcases/extpot/test_extpot.py

```diff
@@ -0,0 +1,103 @@
+#------------------------------------------------------------------------------#
+#  DFTB+: general package for performing fast atomistic simulations            #
+#  Copyright (C) 2006 - 2020  DFTB+ developers group                           #
+#                                                                              #
+#  See the LICENSE file for terms of usage and distribution.                   #
+#------------------------------------------------------------------------------#
+
+'''
+Tests the capability of the interface to
+set an external potential and its gradients.
+'''
+
+
+import numpy as np
+import dftbplus
+from testhelpers import write_autotest_tag
+
+
+LIB_PATH = '../../../../../prog/dftb+/libdftbplus.so'
```

Comment:
The extension of library is particular system dependent, there is no guarantee that it will end on `so`. If the build is performed on OSX we will most probably have a `dynlib` here.

Reply:
The `numpy.ctypeslib.load_library` wrapper for `CDLL` might be worth a try for this purpose, it uses `numpy.distutils.misc_util.get_shared_lib_extension` and `distutils` under the hood to figure out the operating system and select the right extension.

Even better, at least for the test, would be to pass the absolute path to the library from the CMake build system to the tester, since this is the only way to know for sure the right path is actually used.

Reply:
As a Linux-only user, I honestly forgot that compatibility with Win, OSX,... could be a thing. I'm actually not sure at all whether compiling is possible under these operating systems, but maintaining compatibility will definitely not harm.

Thanks for pointing this out, I'll try the suggested Numpy wrapper.